hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -85,4 +91,61 @@ public static HoodieRecord getTaggedRecord(HoodieRecord inputRecord, Option<Hood
     }
     return record;
   }
+
+  /**
+   * Check compatible between new writeIndexType and indexType already in hoodie.properties.
+   * @param writeIndexType new indexType
+   * @param persistIndexType indexType already in hoodie.properties
+   */
+  public static void checkIndexTypeCompatible(IndexType writeIndexType, IndexType persistIndexType) {
+    boolean isTypeCompatible = false;
+    switch (persistIndexType) {
+      case GLOBAL_BLOOM:
+        if (writeIndexType.equals(IndexType.GLOBAL_BLOOM)
+            || writeIndexType.equals(IndexType.BLOOM)
+            || writeIndexType.equals(IndexType.SIMPLE)
+            || writeIndexType.equals(IndexType.GLOBAL_SIMPLE)) {
+          isTypeCompatible = true;
+        }
+        break;
+      case GLOBAL_SIMPLE:
+        if (writeIndexType.equals(IndexType.GLOBAL_SIMPLE)
+            || writeIndexType.equals(IndexType.GLOBAL_BLOOM)) {
+          isTypeCompatible = true;
+        }
+        break;
+      case SIMPLE:
+        if (writeIndexType.equals(IndexType.SIMPLE)) {
+          isTypeCompatible = true;
+        }
+        break;
+      case BLOOM:
+        if (writeIndexType.equals(IndexType.BLOOM)) {
+          isTypeCompatible = true;
+        }
+        break;
+      case INMEMORY:
+        if (writeIndexType.equals(IndexType.INMEMORY)) {
+          isTypeCompatible = true;
+        }
+        LOG.error("PersistIndexType INMEMORY can not be used in production");
+        break;
+      case HBASE:
+        if (writeIndexType.equals(IndexType.HBASE)) {
```

Comment:
if an user is using HBASE then, they could switch to BLOOM, SIMPLE, GLOBAL_BLOOM etc actually